### PR TITLE
NMS-13063: applicationChanged/Created events are missing

### PR DESCRIPTION
**Pull request by mfuhrmann**
### All Contributors

* [x] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
* [x] Have you made a comment in that issue which points back to this PR?
* [x] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [ ] If this is a new or updated feature, is there documentation for the new behavior?
* [ ] If this is new code, are there unit and/or integration tests?
* [ ] If this PR targets a `foundation-*` branch, does it avoid changing files in `$OPENNMS_HOME/etc/`?

### Pull Request Process

One or more reviewers should be assigned to each PR.

If you know that a particular person is subject matter expert in the area your PR affects, feel free to assign one or more reviewers when you create this PR, otherwise reviewers will be assigned for you.

If you have made additions or changes to the documentation, or if you _need_ documentation for these code changes, please make sure a technical writer has looked it over.

Once the reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the Order of the Green Polo) you are welcome to merge the PR.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): https://issues.opennms.org/browse/NMS-13063

### CODE REVIEW

**Comment by mfuhrmann** [comment]
Argh. @RangerRick Should I have picked `release-27.x`?

**Comment by RangerRick** [comment]
Erp, yeah, we shouldn't be changing configs in foundation releases.  Should be safe enough for 27 though, config churn is legal there.  :)

**Comment by mfuhrmann** [comment]
@RangerRick Can we change that here in GH before we merge or do I have to create a new PR?

File: opennms-base-assembly/src/main/filtered/etc/events/opennms.internal.events.xml [hunk 1]
```diff
@@ -248,10 +248,25 @@
...
-      <descr>&lt;p>Application '%parm[applicationName]%' with ID '%parm[applicationId]%' has been deleted.&lt;/p></descr>
-      <logmsg dest="logndisplay">Application '%parm[applicationName]%' with ID '%parm[applicationId]%' has been deleted.</logmsg>
+      <descr>Application '%parm[applicationName]%' with ID '%parm[applicationId]%' has been deleted.</descr>
+      <logmsg dest="logndisplay">Application '%parm[applicationName]%' has been deleted.</logmsg>
       <severity>Warning</severity>
    </event>
+   <event>
+      <uei>uei.opennms.org/internal/applicationChanged</uei>
+      <event-label>OpenNMS-defined node event: applicationChanged</event-label>
+      <descr>The application '%parm[applicationName]%' with ID %parm[applicationId]% has been changed.</descr>
```
Comment (dino2gnt) [inline opennms-base-assembly/src/main/filtered/etc/events/opennms.internal.events.xml:258]: I would single-quote %parm[applicationId]% here for consistency with the existing events

File: opennms-base-assembly/src/main/filtered/etc/events/opennms.internal.events.xml [hunk 2]
```diff
@@ -248,10 +248,25 @@
...
+      <logmsg dest="logndisplay">Application '%parm[applicationName]%' configuration has been changed.</logmsg>
+      <severity>Warning</severity>
+      <operinstruct>Make sure the application '%parm[applicationName]%' still works properly. Please verify it's &lt;a href=&quot;admin/applications.htm?applicationid=%parm[applicationId]%&amp;edit=services&quot;>definition&lt;/a>.</operinstruct>
+   </event>
+   <event>
+      <uei>uei.opennms.org/internal/applicationCreated</uei>
+      <event-label>OpenNMS-defined node event: applicationCreated</event-label>
+      <descr>The application '%parm[applicationName]%' with ID %parm[applicationId]% has been created.</descr>
```
Comment (dino2gnt) [inline opennms-base-assembly/src/main/filtered/etc/events/opennms.internal.events.xml:266]: Here too.

**Review by dino2gnt — CHANGES_REQUESTED** [review]

**Comment by RangerRick** [comment]
> @RangerRick Can we change that here in GH before we merge or do I have to create a new PR?

I changed it, it should do the right thing.

**Comment by mfuhrmann** [comment]
@RangerRick Thanks!

**Review by christianpape — APPROVED** [review]
Ok

File: opennms-base-assembly/src/main/filtered/etc/events/opennms.internal.events.xml [hunk 3]
```diff
@@ -248,10 +248,25 @@
...
+      <logmsg dest="logndisplay">Application '%parm[applicationName]%' has been deleted.</logmsg>
       <severity>Warning</severity>
    </event>
+   <event>
+      <uei>uei.opennms.org/internal/applicationChanged</uei>
+      <event-label>OpenNMS-defined node event: applicationChanged</event-label>
+      <descr>The application '%parm[applicationName]%' with ID '%parm[applicationId]%' has been changed.</descr>
+      <logmsg dest="logndisplay">Application '%parm[applicationName]%' configuration has been changed.</logmsg>
+      <severity>Warning</severity>
+      <operinstruct>Make sure the application '%parm[applicationName]%' still works properly. Please verify it's &lt;a href=&quot;admin/applications.htm?applicationid=%parm[applicationId]%&amp;edit=services&quot;>definition&lt;/a>.</operinstruct>
```
Comment (indigo423) [inline opennms-base-assembly/src/main/filtered/etc/events/opennms.internal.events.xml:261]: What do you mean by making sure the application works properly?

**Review by indigo423 — COMMENTED** [review]
Just minor thing, otherwise looks good to me.

File: opennms-base-assembly/src/main/filtered/etc/events/opennms.internal.events.xml (hunk 3 above)
Comment (mfuhrmann) [reply opennms-base-assembly/src/main/filtered/etc/events/opennms.internal.events.xml:261]: Maybe a bit confusing, right. People should get a hint to check the configuration after a change. Maybe now it does not monitor anymore as expected or initially designed.

**Comment by mfuhrmann** [comment]
giphy

**Comment by mfuhrmann** [comment]
Closing because of commit issue https://github.com/OpenNMS/opennms/pull/3314/commits/7d972d777283a40e1e2678f1211b2cf991ee22f4

Creating a new PR

**Pull request closed without merging**